### PR TITLE
#2266 Create projects using new shipyard spec

### LINKED ISSUE #2266: CLI and shipyard-service support creating a project using the new shipyard spec
**Tasks:**
- Use CRUD operations for creating new projects, e.g. the [current endpoint](https://github.com/keptn/keptn/blob/2b876143ed6e7696322560e301ec8328a42223db/api/handlers/project.go#L27) should not send-out a cloud-event. This endpoint should forward the request to the shipyard-controller.
- The shipyard-controller should provide HTTP endpoints for creating and deleting projects/services. This endpoint will be responsible for creating the entity for the project within the configuration service and uploading the secret. It will also send  a `sh.keptn.event.project.create.finished` event (allows other services to register on those event). This event must not contain the git credentials for the project. Also, instead of passing on the git credentials to the configuration service, it should directly create the K8s secret for the credentials. The logic for creating/deleting projects/services is currently located in the [shipyard-service](https://github.com/keptn/keptn/blob/2b876143ed6e7696322560e301ec8328a42223db/shipyard-service/main.go#L174) and can be removed there. 
- Instead of opening a WebSocket communication, the result should be returned directly in the HTTP response from the shipyard controller
- Remove the Shipyard service when all of its current responsibilities have been moved to the shipyard controller
- The CLI should accept the new shipyard format for creating new projects
- The CLI should use the new endpoint of the shipyard-controller for creating projects and services
- The shipyard-controller should accept the new [shipyard format](https://github.com/keptn/go-utils/blob/4641d285f58fa652db8ba7f4dcfd3184bb49847d/pkg/lib/v0_2_0/shipyard.go#L6) for creating a new project
- Neither the CLI nor the shipyard-controller should accept the old shipyard format (however, we will need an upgrade path)
- If a user specifies an old shipyard, the CLI references the user to the new shipyard spec.

**Definition of Done:**
- [ ] The Keptn CLI can be used to create a new project with multiple stages

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2266 

This PR adds new endpoints in the shipyard-controller for creating/deleting projects and services.
The endpoints for creating/deleting projects/services have been removed from the api-service

Instead of sending a project.create/service.create event, the shipyard controller directly creates all the required entities via the configuration-service and gives feedback in the HTTP response.

The shipyard-service has been removed, since all of its responsibilities are now handled in the shipyard-controller.

The specification of the new endpoints in the shipyard-controller is as follows:

swagger-shipyard

Screenshot 2020-10-22 at 08 25 11

Screenshot 2020-10-22 at 08 25 33

Screenshot 2020-10-22 at 08 25 49

Screenshot 2020-10-22 at 08 26 06

Screenshot 2020-10-22 at 08 26 21

### CODE REVIEW

File: cli/cmd/delete_project.go [hunk 1]
```diff
@@ -87,32 +85,25 @@ var delProjectCmd = &cobra.Command{
...
 						}
-
-						// if eventContext is available, open WebSocket communication
-						if eventContext != nil && !SuppressWSCommunication {
-							err := websockethelper.PrintWSContentEventContext(eventContext, endPoint)
-							if err != nil {
-								fmt.Println("Could not delete service " + service.ServiceName + ", but continuing with project deletion.")
-							}
-						}
+						fmt.Println("Project deleted successfully")
```
Comment (bacherfl) [inline cli/cmd/delete_project.go:93]: use logging.PrintLog()

File: cli/cmd/update_project.go [hunk 2]
```diff
@@ -70,30 +68,26 @@ For more information about updating projects or upstream repositories, please go
...
 			if err != nil {
 				return fmt.Errorf("Update project was unsuccessful. %s", *err.Message)
 			}
 
-			// if eventContext is available, open WebSocket communication
-			if eventContext != nil && !SuppressWSCommunication {
-				return websockethelper.PrintWSContentEventContext(eventContext, endPoint)
-			}
-			logging.PrintLog("Project updated successful", logging.InfoLevel)
+			logging.PrintLog("Project updated successfuly", logging.InfoLevel)
```
Comment (bacherfl) [inline cli/cmd/update_project.go:90]: typo: successfully

File: shipyard-controller/api/project.go [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+		namespace := project + "-" + stage.Name
+		msg += fmt.Sprintf("- A potentially created namespace %s is not managed by Keptn anymore but is not deleted. "+
+			"If you would like to delete this namespace, please execute "+
+			"'kubectl delete ns %s'\n", namespace, namespace)
+	}
+	return strings.TrimSpace(msg)
+}
+
+func (pm *projectManager) updateProject(params *operations.CreateProjectParams) error {
+	pm.logger.Info(fmt.Sprintf("checking if project %s already exists before creating it", *params.Name))
```
Comment (bacherfl) [inline shipyard-controller/api/project.go:269]: check if the project exists

File: shipyard-controller/api/project_test.go [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "invalid entity name",
+			args: args{
+				createProjectParams: &operations.CreateProjectParams{
+					GitRemoteURL: "",
```
Comment (bacherfl) [inline shipyard-controller/api/project_test.go:349]: remove empty strings from test objects

**Review by warber — DISMISSED** [review]
lgtm

**Review by warber — DISMISSED** [review]
lgtm

**Review by johannes-b — APPROVED** [review]
lgtm

**Pull request merged**